### PR TITLE
Integrate /problems with datastore

### PR DESCRIPTION
**Pull request by ifeomi**
- Branch is based off of abstract_datastore branch #24 
- Queries for all problems, renders route with result of query, and links to `problems/:id`
image

### CODE REVIEW

File: html/problems.hbs [hunk 1]
```diff
@@ -3,7 +3,7 @@
     {{#each problems}}
     <div class="card">
       <div class="card-body">
-        <h5 class="card-title">{{this.title}}</h5>
+        <a href="problems/{{this.id}}"><h5 class="card-title">{{this.title}}</h5></a>
```
Comment (nicholashusin) [inline html/problems.hbs]: Probably want to use absolute link just to be safe

**Review by lalaAdasia00 — APPROVED** [review]
It looks good :)

**Review by nicholashusin — APPROVED** [review]
LGTM :)

File: html/problems.hbs (hunk 1 above)
Comment (ifeomi) [reply html/problems.hbs]: done, ty!

File: routes/problems.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    this.id = id;
+    this.title = title;
+    this.text = text;
+  }
+}
+
+module.exports = function(app) {
+  app.get('/problems', async function(request, response) {
+    const query = datastore.createQuery(PROBLEMS_KIND);
+    const [entities] = await datastore.runQuery(query);
```
Comment (chronologos) [inline routes/problems.js:25]: same comment here as for Nick's pull request - pagination is important when the API may return multiple pages.

Comment (ifeomi) [reply routes/problems.js:25]: Right now, this route just queries for all problems. I added a todo to add a limit to the query and support pagination post-MVP.

**Review by chronologos — APPROVED** [review]

**Pull request merged**